Improve Parameter Resolver documentation (#73)
mulesoft/docs-mule-sdk: modules/ROOT/pages/special-parameters.adoc

[[parameter-resolver]]
== ParameterResolver<Type>

> `ParameterResolver<Type>` allows you to defer the resolution of an expression on a Operation. This is useful when a parameter that takes an expression as a value is not required. You can improve performance by resolving such an expression on demand from within the connectors's code instead of resolving it automatically.

Reviewer: ```suggestion
`ParameterResolver<Type>` allows you to defer the resolution of an expression on an operation. This is useful when a parameter that takes an expression as a value is not required. You can improve performance by resolving such an expression on demand from within the connectors's code instead of resolving it automatically.
```

Compared to <<literal, `Literal<Type>`>> parameters, a `ParameterResolver<Type>` parameter with an expression continues to be treated as a dynamic parameter whether the value is resolved or not.

> WARNING: `ParameterResolver` is not compatible to be used inside Configurations or Connection Providers.

Reviewer: ```suggestion
NOTE: You cannot use `ParameterResolver` inside of configurations or connection providers.
```
